integrations/docs/API/swagger.yaml

```diff
@@ -0,0 +1,287 @@
+          schema:
+            type: integer
+            format: int32
+            minimum: 0
+            maximum: 50
+      responses:
+        '200':
+          description: search results matching criteria
+          content:
+            application/json:
+              schema:
+                type: array
+                items:
+                  $ref: '../../../schema/system/integration.schema'
+        '400':
+          description: bad input parameter
+    post:
+      tags:
+        - admins
+      summary: adds an Integration item
+      operationId: addIntegration
+      description: |
+       Adds an Integration to the repository, expecting the internal URLs to be accessible
+      requestBody:
+        required: true
+        description: Inventory item to add
+        content:
+          application/json:
+            schema:
+              $ref: '../../../schema/system/integration.schema'
+      responses:
+        '201':
+          description: item created
+        '400':
+          description: 'invalid input, object invalid'
+        '409':
+          description: an existing item already exists
+  /store:
+    get:
+      tags:
+        - developers
+      summary: get all stored Integrations
+      operationId: searchStore
+      description: |
+        By passing in the appropriate search attributes, you can search for
+        available Integrations in the store
+      parameters:
+        - in: query
+          name: searchString
+          description: pass an optional search string for looking up Integration in the store
+          required: false
+          schema:
+            type: string
+        - in: query
+          name: limit
+          description: maximum number of records to return
+          schema:
+            type: integer
+            format: int32
+            minimum: 0
+            maximum: 50
+      responses:
+        '200':
+          description: search results matching criteria
+          content:
+            application/json:
+              schema:
+                type: array
+                items:
+                  $ref: '../../../schema/system/integration-instance.schema'
+        '400':
+          description: bad input parameter
+    post:
+      tags:
+        - admins
+      summary: adds an Integration item to store
+      operationId: storeIntegration
+      description: |
+       Adds an Integration to the system store, expecting the internal URLs to be accessible
+      requestBody:
+        required: true
+        description: Inventory item to add
+        content:
+          application/json:
+            schema:
+              $ref: '../../../schema/system/integration-instance.schema'
+      responses:
+        '201':
+          description: item created
+        '400':
+          description: 'invalid input, object invalid'
+        '409':
+          description: an existing item already exists
+  /store/{id}/status:
```

Comment:
we will need a way to bulk check status - or else we will need to make one call for each added api to check status - which may be too much